drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/columns/ValueListUiColumn.java

```diff
@@ -28,79 +28,104 @@
-                                                       final GridBodyCellRenderContext context ) {
-                       t.setText( getLabel( value.toString() ) );
-                   }
+class ValueListUiColumnCellRenderer
+        extends BaseSingletonDOMElementUiColumn.CellRenderer<String, ListBox, ListBoxDOMElement<String, ListBox>> {
 
-                   private String getLabel( final String value ) {
-                       if ( !isMultipleSelect ) {
-                           return valueListLookup.get( value );
-                       }
-                       final StringBuilder sb = new StringBuilder();
-                       final String[] values = value.split( "," );
-                       for ( int i = 0; i < values.length; i++ ) {
-                           if ( i == 0 ) {
-                               sb.append( valueListLookup.get( values[ i ] ) );
-                           } else {
-                               sb.append( "," ).append( valueListLookup.get( values[ i ] ) );
-                           }
-                       }
-                       return sb.toString();
-                   }
+    private final Map<String, String> valueListLookup;
+    private boolean isMultipleSelect;
 
-               },
-               width,
-               isResizable,
-               isVisible,
-               access,
-               factory );
-        this.valueListLookup.putAll( valueListLookup );
+    public ValueListUiColumnCellRenderer(final SingletonDOMElementFactory<ListBox, ListBoxDOMElement<String, ListBox>> factory,
+                                         final Map<String, String> valueListLookup,
+                                         final boolean isMultipleSelect) {
+        super(factory);
+        this.valueListLookup = valueListLookup;
+        this.isMultipleSelect = isMultipleSelect;
     }
 
     @Override
-    public void doEdit( final GridCell<String> cell,
-                        final GridBodyCellRenderContext context,
-                        final Callback<GridCellValue<String>> callback ) {
-        factory.attachDomElement( context,
-                                  CallbackFactory.makeOnCreationCallback( factory,
-                                                                          cell,
-                                                                          valueListLookup ),
-                                  CallbackFactory.makeOnDisplayListBoxCallback() );
+    protected void doRenderCellContent(final Text t,
+                                       final String value,
+                                       final GridBodyCellRenderContext context) {
+        t.setText(getLabel(value.toString()));
     }
 
+    private String getLabel(final String value) {
+        if (!isMultipleSelect) {
+            return getValue(value);
+        }
+        final StringBuilder sb = new StringBuilder();
+        final String[] values = value.split(",");
```

Comment:
I know it's old code but should I be worried about the split here?

Reply:
Double checking, perhaps the fix is not the best.

Reply:
@manstis Ok, I think this is the best way to do it. Unless you know why the list is empty when the value is list. For example with a cell value "a,b" the a or b can not be found from the `valueListLookup` map.